### PR TITLE
migrate project to @roxi/routify

### PR DESCRIPTION
**Pull request by Schparky**
Issues encountered/solved:
- It was difficult to set up routify individually, so I pulled in the rollup config from Translation Friends, which included the service worker generation.
- Routify requires a much different project layout, so many files moved in this process.
- Routify `url`, `params`, and `goto` cannot be used outside of a Svelte file or in the App.svelte file.
- As a result of the above limitation, much of App.svelte was moved to the top _layout.svelte file
- Used the tokens.js from TF.

### CODE REVIEW

File: src/components/CreateOrUpdateRequest.svelte [hunk 1]
```diff
@@ -84,9 +81,9 @@ function imageUploaded(event) {
 }
 
 async function cancelRequest() {
-  await cancel(params.id)
+  await cancel($params.id)
```
Comment (hobbitronics) [inline src/components/CreateOrUpdateRequest.svelte:84]: Should this be `await cancel($params.requestId)`?

Comment (Schparky) [reply src/components/CreateOrUpdateRequest.svelte:84]: Yes, good catch!

File: src/pages/login.svelte [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<script>
+  import Login from '../components/Login.svelte'
+  </script>
```
Comment (hobbitronics) [inline src/pages/login.svelte:3]: indentation is off

**Review by hobbitronics — APPROVED** [review]
Looks like this was a lot of work. Way to power through it.

File: src/components/Messages.svelte [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-import { replace } from 'svelte-spa-router'
+import { params, redirect } from '@roxi/routify'
 import { conversations } from '../data/messaging'
 
-export let params = {} // URL path parameters, provided by router.
-
 let conversationId
 
-$: conversationId = params.id
+$: conversationId = $params.messageId
```
[the remark this answers is not in the document]
Comment (Schparky) [reply src/components/Messages.svelte]: Agreed, it is confusing. It's actually the conversation ID. I renamed it as such.

File: src/data/filtering.js [hunk 4]
```diff
@@ -102,7 +103,7 @@ export function removeFilter(name) {
  * @param {Object} updates
  */
 export function setFilters(updates) {
-  updateQueryString(updates)
+  return updateQueryString(updates)
```
[the remark this answers is not in the document]
Comment (Schparky) [reply src/data/filtering.js:107]: yep

File: src/data/gqlQueries.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ export async function getUser() {
     }
   }`)
 
-  return response.user || {}
+  return response?.user || {}
```
[the remark this answers is not in the document]
Comment (Schparky) [reply src/data/gqlQueries.js:10]: Yep, @hobbitronics found it a few weeks ago.

File: src/pages/_layout.svelte [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  import { page } from '@roxi/routify'
+
+  const publicRoutes = ['/terms', '/privacy', '/join']
+
+  let isPublicRoute = false
+
+  $: isPublicRoute = publicRoutes.some(publicRoute => $page.path.startsWith(publicRoute))
+  $: ! (isPublicRoute || isUserAuthn()) && authenticate() // should only react to location changes, not user changes.
+  $: userIsAuthn = $me.id
+  $: minimal = $page.path.startsWith('/welcome') || ! userIsAuthn
```
[the remark this answers is not in the document]
Comment (Schparky) [reply src/pages/_layout.svelte]: Thanks

**Review by forevermatt — APPROVED** [review]

**Pull request merged**